domain/src/main/java/com/thoughtworks/go/domain/materials/git/RefSpecHelper.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.domain.materials.git;
+
+import com.thoughtworks.go.util.command.UrlArgument;
+
+public class RefSpecHelper {
```

Comment:
This util class handles all the logic to extract meaningful source and destination refs from refspecs, as well as defaulting to old behaviors when provided just a plain branch.

This util class exists since the logic is shared between `GitMaterial` and `GitCommand`.